src/reading.rs

```diff
@@ -528,19 +528,167 @@ impl BasePacketReader {
 	}
 }
 
+// TODO replace 'a with GAT as soon as they are stabilized
```

Comment:
Is there any benefit in using GATs here? Also, ogg does have MSRV guarantees, so stabilization is not the event that would make us pick up a feature, at least not immediately.

Reply:
GATs would allow to annotate the lifetime directly at the `Buf` type instead of the trait, which would make the interface a bit nicer. I removed the TODO for now, I actually just wrote it out of habit. You can decide however you want to proceed in this regard. :)